src/haddock/gear/config_reader.py

```diff
@@ -275,9 +288,23 @@ class MultilineListDefinitionError(Exception):
 
 
 # main public API
-def read_config(f):
-    """Parse HADDOCK3 config file to a dictionary."""
-    with open(f, 'r') as fin:
+def read_config(fpath):
```

Comment:
I am not a super fan of these nested methods with fifteen lines of documentation and a single line of code..why do we have to define _read_config? Can't we incorporate it in read_config(fpath)? I think it will improve the code readability. If not related to this PR feel free to open another issue :)

Reply:
The reason for this is to decouple the open file logic from the parser logic. As simple as that. Having the logic to read from disk coupled with the logic to parse the fail makes testing and debugging more difficult. That's why I opt for this strategy. To have a public API that only reads the file and a private API that takes care of the actual parsing. In the future, if we need to use the parsing logic to parse piped data (that does not require opening the file), we can use the parsing function directly without needing to rewrite anything else.